src/freenet/pluginmanager/PluginManager.java

```diff
@@ -1064,65 +975,35 @@ public String handleHTTPPost(String plugin, HTTPRequest request) throws PluginHT
 	}
 
 	public void killPlugin(String name, long maxWaitTime, boolean reloading) {
-		PluginInfoWrapper pi = null;
-		boolean found = false;
-		synchronized(pluginWrappers) {
-			for(int i = 0; i < pluginWrappers.size() && !found; i++) {
-				pi = pluginWrappers.get(i);
-				if(pi.getThreadName().equals(name)) {
-					found = true;
-					break;
-				}
+		for (PluginInfoWrapper pluginInfoWrapper : pluginWrappers) {
+			if (pluginInfoWrapper.getThreadName().equals(name)) {
```

Comment:
Why did you remove the break after stopping the plugin? I don't see why we should keep looking through threads.